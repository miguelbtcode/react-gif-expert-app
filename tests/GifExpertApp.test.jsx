import { render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {
  test('Debe hacer match con el snapshoot', () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });

  test('Debe mostrar el título de GifExpertApp', () => {
    render(<GifExpertApp />);

    const heading = screen.getAllByRole('heading', { level: 1 })[0];

    expect(heading.textContent).toBe('GifExpertApp');
  });

  test('Debe mostrar las categorías iniciales', () => {
    render(<GifExpertApp />);

    // Verifica que hay exactamente 2 categorías en el DOM
    const categories = screen.getAllByRole('heading', { level: 3 });

    expect(categories.length).toBe(2);
    expect(categories[0].textContent).toBe('One Punch');
    expect(categories[1].textContent).toBe('Dragon Ball');
  });
});
