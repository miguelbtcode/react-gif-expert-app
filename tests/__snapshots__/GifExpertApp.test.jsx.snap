// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe hacer match con el snapshoot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifts"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
  <h3>
    Dragon Ball
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
